### DISCUSSION TITLE
the server responded with a status of 413 (Payload Too Large)

### DISCUSSION THREAD

**Discussion by xm0809** [lowdefy/lowdefy#1009] [Questions and Help]
A problem “Failed to load resource: the server responded with a status of 413 (Payload Too Large)”   occurred，when  calling  a request.Maybe we need  increase   the file size limit about " client_ max_ body_ Size" or "LimitRequestBody" .  But,how to set the parameter about lowdefy web server?

**Comment by SamTolmay** [comment]
Hi @xm0809 

We are experiencing this issue because in version 3, the `event`, `input`, `global`, `state`, and `urlQuery` data objects are all being sent with each request, so that you can use them in requests and connections.

We have already increased the maximum payload size in v3 to 5mb to help with this. Maybe we can make this configurable, but it is already quite large.

We have changed this in version 4, we have created a new payload object, which is specified in the configuration, so that you can decide what data to send to the server. Please see #641 for details. Unfortunately this is a breaking change, so we cannot release this fix in version 3.

One way to resolve this error on v3 is to make sure your state and global objects do not get too large. For example, if you have a table, instead of setting the data to state you can read from the request directly:
```yaml
id: my_table
type: AgGridAlpine
properties:
  rowData:
    _request: my_request
```

Or you can try paginating the data.

We are working on getting version 4 ready as soon as possible so that we can fix this properly.

Please get in touch with any specific questions on how you can reduce the size of data in state.
